.github/workflows/nomad-requirements.yml

```diff
@@ -0,0 +1,56 @@
+name: Validate Dependencies Compatibility
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    # Run workflow only when there are changes in pyproject.toml or dev-requirements.txt
+    # paths:
+    # - 'pyproject.toml'
+    # - 'dev-requirements.txt'
+
+jobs:
+  validate_dependencies:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout pynxtools
+      uses: actions/checkout@v2
+
+    - name: Checkout NOMAD from GitLab
+      run: |
+        git clone https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR.git nomad
+        cd nomad
+        git checkout develop
+        git submodule update --init --recursive
+    
+    - name: Replace pynxtools dependency in NOMAD pyproject.toml
+      working-directory: ./nomad
+      run: |
+        sed -i 's|pynxtools\[convert\]==[0-9]\+\(\.[0-9]\+\)\{0,2\}|pynxtools\[convert\]@git+https://github.com/FAIRmat-NFDI/pynxtools.git@${{ github.head_ref }}|' pyproject.toml
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Install uv
+      run: |
+        curl -LsSf https://astral.sh/uv/install.sh | sh 
+
+    - name: Generate (dev-)requirements.txt from modified pyproject.toml
+      working-directory: ./nomad
+      run: |
+        uv pip compile -p 3.9 --annotation-style=line --extra=infrastructure --extra=parsing --output-file=requirements.txt dependencies/nomad-dos-fingerprints/pyproject.toml dependencies/parsers/eelsdb/pyproject.toml pyproject.toml
+        uv pip compile -p 3.9 --annotation-style=line --extra=dev --extra=infrastructure --extra=parsing --output-file=requirements-dev.txt requirements.txt pyproject.toml
+
+    - name: Install NOMAD dependencies with pynxtools from current branch
```

Comment:
I think strictly we don't really need to install it in the end. The most important part is that we can generate the dependencies and we might not have much benefit installing it without testing it.

Reply:
I left it here because maybe we will want to do something else in this nomad workflow (like checking that nomad parses nexus data properly with any changes to pynxtools).